src/domains/issue/service/issue.service.impl.ts

```diff
@@ -506,4 +525,28 @@ export class IssueServiceImpl implements IssueService {
       }
     });
   }
+
+  /**
+   * Handles the change log entry for an issue based on the provided input.
+   * If the 'to' state is defined, it updates the corresponding field of the issue.
+   * @param input An object containing the details of the change event, including the field to update, the issue ID, the project ID, and the new value (to).
+   * @returns A promise that resolves once the issue has been updated.
+   */
+  private async handleIssueChangeLogByField(input: ChangeScalarEventInputExtended): Promise<void> {
```

Comment:
Code's kinda hard to read. I'd suggest something more like this:
```typescript
const field = input.field
    switch (field) {
      case 'assignee': {
        const asignee: UserDTO | null = input.to != null ? await this.userRepository.getByEmail(input.to) : null;
        await this.issueRepository.updateById({ issueId: input.issueId, assigneeId: asignee?.id });
        break;
      }
      case 'stage': {
        const project: ProjectDTO = await this.validateProjectExistence(input.projectId);
        const projectStage = input.to != null ? await this.projectStageRepository.getByProjectIdAndName({ projectId: project.id, name: input.to }) : null;
        await this.issueRepository.updateById({ issueId: input.issueId, projectStageId: projectStage?.id });
        break;
      }
      default:
        throw new Error('Malformed field.')
    }
```

Reply:
corrected